### PR TITLE
When no upstream in upstream_list is specified, list all of them

### PR DESCRIPTION
**Pull request by CallMeFoxie**
Since we were interested in that option as well as others, I've taken the liberty to code it.

(Issue #46 for example)

Cheers,
Ashley

### CODE REVIEW

File: src/ngx_http_upsync_module.c [hunk 1]
```diff
@@ -3676,6 +3676,40 @@ ngx_http_upsync_set(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
     return NGX_CONF_OK;
 }
```
Comment (xiaokai-wang) [inline src/ngx_http_upsync_module.c]: It should be two black lines between functions.

File: src/ngx_http_upsync_module.c [hunk 2]
```diff
@@ -3676,6 +3676,40 @@ ngx_http_upsync_set(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
...
 
+static void ngx_http_upsync_show_upstream(ngx_http_upstream_srv_conf_t *uscf, ngx_buf_t *b) {
+    ngx_http_upstream_rr_peers_t    *peers = NULL;
+    ngx_uint_t                       i;
+    ngx_str_t                       *host;
+
```
Comment (xiaokai-wang) [inline src/ngx_http_upsync_module.c]: In my style, the lines would be:
 ngx_str_t                                       *host;
 ngx_uint_t                                      i;
 ngx_http_upstream_rr_peers_t    *peers = NULL;
Any convenience, making a change will be good.

File: src/ngx_http_upsync_module.c [hunk 3]
```diff
@@ -3676,6 +3676,40 @@ ngx_http_upsync_set(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
...
+    }
+
+    b->last = ngx_snprintf(b->last, b->end - b->last, "Upstream name: %V; ",
+            host);
+    b->last = ngx_snprintf(b->last, b->end - b->last,
+            "Backend server count: %d\n", peers->number);
+
+    for (i = 0; i < peers->number; i++) {
+        b->last = ngx_snprintf(b->last, b->end - b->last, "        server %V",
+                &peers->peer[i].name);
```
Comment (xiaokai-wang) [inline src/ngx_http_upsync_module.c]: Nginx's style is: 
b->last = ngx_snprintf(b->last, b->end - b->last, "        server %V",
                                    &peers->peer[i].name);

File: src/ngx_http_upsync_module.c [hunk 4]
```diff
@@ -3676,6 +3676,40 @@ ngx_http_upsync_set(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
...
+        b->last = ngx_snprintf(b->last, b->end - b->last, " weight=%d",
+                peers->peer[i].weight);
```
Comment (xiaokai-wang) [inline src/ngx_http_upsync_module.c]: It cannt show well, like that you delete.

File: src/ngx_http_upsync_module.c [hunk 5]
```diff
@@ -3676,6 +3676,40 @@ ngx_http_upsync_set(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
...
+                peers->peer[i].fail_timeout);
+
+        if (peers->peer[i].down) {
+            b->last = ngx_snprintf(b->last, b->end - b->last, " down");
+        }
+
+        b->last = ngx_snprintf(b->last, b->end - b->last, ";\n");
+    }
+}
+
```
Comment (xiaokai-wang) [inline src/ngx_http_upsync_module.c]: Two black lines.

**Comment by xiaokai-wang** [comment]
Sure, it's great.

**Comment by xiaokai-wang** [comment]
Please merge three commits into one and make a new pr, thanks.

**Comment by CallMeFoxie** [comment]
I am not sure what you mean at the 
&peers->peer[i].name
and
peers->peer[i].weight
lines.
They are the same as they were in your original code (after my latest commit to fix the code style).

**Comment by xiaokai-wang** [comment]
please ignore that, it's a little delay.

please merge three commits into one and make a new pr.

**Pull request closed without merging**